chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -818,53 +818,89 @@ impl PeerManagerActor {
                     NetworkResponses::RouteNotFound
                 }
             }
-            NetworkRequests::StateRequestHeader { shard_id, sync_hash, peer_id } => {
-                if self.state.tier2.send_message(
-                    peer_id,
-                    Arc::new(PeerMessage::StateRequestHeader(shard_id, sync_hash)),
+            NetworkRequests::StateRequestHeader { shard_id, sync_hash, sync_prev_prev_hash } => {
+                // The node needs to include its own public address in the request
+                // so that the response can be sent over a direct Tier3 connection.
+                let Some(addr) = *self.state.my_public_addr.read() else {
+                    return NetworkResponses::AwaitingIpSelfDiscovery;
+                };
```

Comment:
Does it actually mean awaiting or can it mean that the self discovery failed? Is there some background loop that tries self discovery until it succeeds?

Reply:
I guess not really, it can take a small amount of time when the node starts to complete it successfully. But if it doesn't work it won't fix itself later. I modified the response name for clarity.